index.bs

```diff
@@ -101,21 +118,190 @@ spec: webidl
+      dictionary DevicePermissionDescriptor : PermissionDescriptor {
+        DOMString deviceId;
+      };
+    </pre>
+    <p>
+      General access to cameras is represented by `{name: 'camera'}`, while
+      access to a particular camera is represented by `{name: 'camera',
+      deviceId: cameraId}`.
+    </p>
+  </div>
+</section>
+<section>
+  <h2 id="permission-operations">Permission states</h2>
+  <p>
+    The user agent is responsible for tracking what powerful features each
+    <a>realm</a> has the user's permission to use. Other specifications can use
+    the operations defined in this section to retrieve the UA's notion of what
+    permissions are granted or denied, and to ask the user to grant or deny more
+    permissions.
+  </p>
+
+  <p>
+    Other specifications can also add more constraints on the UA's behavior in
+    these algorithms.
+  </p>
+
+  <p>
+    Within this section, |descriptor| is an instance of the <a>permission
+    descriptor type</a> of the powerful feature named by
+    <code>|descriptor|.{{PermissionDescriptor/name}}</code>.
+  </p>
+
+  <section>
+    <h3 id="reading-current-states">Reading the current permission state</h3>
+    <p>
+      |descriptor|'s <dfn export>permission state</dfn> is one
+      of {{"granted"}}, {{"prompt"}}, or {{"denied"}}, indicating respectively
+      if the calling algorithm should succeed without prompting the user, show
+      the user a prompt to decide whether to succeed, or fail without prompting
+      the user. The UA must return whichever of these values most accurately
+      reflects the user's intent. Successive uses of |descriptor|'s <a>permission
+      state</a> with the same <a>current settings object</a> should return the
+      same value, unless the UA receives new information about the user's
+      intent.
+    </p>
+
+    <p class="issue" id="issue-current-entry-incumbent-or-relevant">
+      Safari is the only known UA that returns different results from this
+      algorithm for different settings objects with the same origin. We should
+      test which of the <a
+      href="https://html.spec.whatwg.org/multipage/webappapis.html#realms-settings-objects-global-objects">several
+      possible settings objects</a> it uses.
+    </p>
+
+    <p>
+      Some powerful features have more information associated with them than
+      just a {{PermissionState}}. For example, {{MediaDevices/getUserMedia()}}
+      needs to determine <em>which</em> cameras the user has granted the
+      <a>current realm</a> permission to access. Each of these features defines
+      an <a>extra permission data type</a>, and then a {{PermissionName}} |name|'s
+      <dfn export>extra permission data</dfn> is the instance of that type
+      matching the UA's impression of the user's intent. Successive uses of
+      |name|'s <a>extra permission data</a> should return the same value, unless
+      the UA receives new information about the user's intent.
```

Comment:
the text " unless the UA receives new information about the user's intent" seems to be repeated throughout. Can we capture this as a concept to avoid having to repeat it?